tests_integ/helpers/deployer/exceptions/exceptions.py

```diff
@@ -0,0 +1,93 @@
+"""
+Exceptions that are raised by sam deploy
+"""
+import click
+
+
+class UserException(click.ClickException):
+    """
+    Base class for all exceptions that need to be surfaced to the user. Typically, we will display the exception
+    message to user and return the error code from CLI process
+    """
+
+    exit_code = 1
+
+    def __init__(self, message, wrapped_from=None):
+        self.wrapped_from = wrapped_from
+
+        click.ClickException.__init__(self, message)
+
+
+class ChangeEmptyError(UserException):
+    def __init__(self, stack_name):
+        self.stack_name = stack_name
+        message_fmt = "No changes to deploy. Stack {stack_name} is up to date"
+        super(ChangeEmptyError, self).__init__(message=message_fmt.format(stack_name=self.stack_name))
+
+
+class ChangeSetError(UserException):
+    def __init__(self, stack_name, msg):
+        self.stack_name = stack_name
+        self.msg = msg
+        message_fmt = "Failed to create changeset for the stack: {stack_name}, {msg}"
+        super(ChangeSetError, self).__init__(message=message_fmt.format(stack_name=self.stack_name, msg=self.msg))
+
+
+class DeployFailedError(UserException):
+    def __init__(self, stack_name, msg):
+        self.stack_name = stack_name
+        self.msg = msg
+
+        message_fmt = "Failed to create/update the stack: {stack_name}, {msg}"
+
+        super(DeployFailedError, self).__init__(message=message_fmt.format(stack_name=self.stack_name, msg=msg))
+
+
+class GuidedDeployFailedError(UserException):
```

Comment:
Can we consider trimming errors like this that are SAM CLI specific down to only errors relevant to us?

Reply:
Yes, removing the ones we don't use